port_ocean/clients/port/mixins/entities.py

```diff
@@ -58,11 +58,19 @@ async def upsert_entity(
             )
         handle_status_code(response, should_raise)
         result = response.json()
-        result_entity = Entity.parse_obj(result)
-        # Set the results of the search relation and identifier to the entity
-        entity.identifier = result_entity.identifier or entity.identifier
-        entity.relations = result_entity.relations or entity.relations
-        return entity
+        result_entity = (
+            Entity.parse_obj(result["entity"]) if result.get("entity") else entity
```

Comment:
returning all the result entity means that we return the full entity that is inside port, this means that if we just ingested the relations of an entity but it has 20 more properties it will be returned here and passed forward to the resync phase. This can be problematic from a memory point of view and that is why we only took the identifier and the relations from the result entity and not the full entity.

Reply:
we need the blueprint as well to delete the identifier of a specific blueprint only as there may be entities with same identifier but different blueprints